torch/distributions/transformed_distribution.py

```diff
@@ -69,7 +69,17 @@ def expand(self, batch_shape, _instance=None):
 
     @constraints.dependent_property(is_discrete=False)
     def support(self):
-        return self.transforms[-1].codomain if self.transforms else self.base_dist.support
+        if not self.transforms:
+            return self.base_dist.support
+        support = self.transforms[-1].codomain
+        # Adjust event_dim by maximum among transforms.
+        event_dim = self.base_dist.support.event_dim
+        for t in self.transforms:
+            event_dim = max(event_dim, t.domain.event_dim)
+            event_dim += t.codomain.event_dim - t.domain.event_dim
```

Comment:
This should belong to the computation of `event_dim` in the constructor.

Reply:
Good point! Actually the constructor needs to do a full shape computation, not merely a dim computation. Thus I believe we will need to implement `Transform.forward_shapes()`. I'll add that...

Reply:
Agreed!

Reply:
Done.